deploy-service/universal/src/test/java/com/pinterest/teletraan/universal/security/BasePastisAuthorizerTest.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2024 Pinterest, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.pinterest.teletraan.universal.security;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.pinterest.commons.pastis.PastisAuthorizer;
+import com.pinterest.teletraan.universal.security.bean.AuthZResource;
+import com.pinterest.teletraan.universal.security.bean.ServicePrincipal;
+import com.pinterest.teletraan.universal.security.bean.UserPrincipal;
+import java.io.IOException;
+import java.util.Arrays;
+import javax.ws.rs.container.ContainerRequestContext;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+class BasePastisAuthorizerTest {
+    private static final String principalName = "testUser";
+    private static final String spiffeId = "testSpiffe";
+    private static final String group = "testGroup";
+    private static final String actionRead = "READ";
+    private static final String actionWrite = "WRITE";
+    private static final String requestTemplate =
+            "{\"input\":{\"principal\":{\"id\":\"%s\",\"type\":\"%s\",\"groups\":[\"%s\"]},\"action\":\"%s\",\"resource\":{\"name\":\"%s\",\"type\":\"%s\"}}}";
```

Comment:
[Optional] Please consider to replace hardcoded string with HashMap which converts to JSON by using ObjectMapper from Jackson. It will improve readability of the test because currently the `\` symbol is unnecessary as for me.

Reply:
I agree there is likely a better way to reformat this. However the correctness can't be guaranteed if we use the same mechanism to validate a mechanism. In this case the string is generated by `ObjectMapper`, I don't think we should validate the outcome using `ObjectMapper` again.